quickwit/quickwit-indexing/src/actors/uploader.rs

```diff
@@ -132,15 +144,22 @@ impl Handler<PackagedSplitBatch> for Uploader {
         ctx: &ActorContext<Self>,
     ) -> Result<(), ActorExitStatus> {
         fail_point!("uploader:before");
-        let (split_uploaded_tx, split_uploaded_rx) =
-            oneshot::channel::<SequencerCommand<SplitUpdate>>();
-
-        // We send the future to the sequencer right away.
-
-        // The sequencer will then resolve the future in their arrival order and ensure that the
-        // publisher publishes splits in order.
-        ctx.send_message(&self.sequencer_mailbox, split_uploaded_rx)
-            .await?;
+        let uploader_sender = match &self.split_update_mailbox {
+            SplitUpdateMailbox::Sequencer(sequencer_mailbox) => {
```

Comment:
this is not the right place to put this code. This should be in the `impl SplitUpdateMailbox`